sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/Triggers.scala

```diff
@@ -17,13 +17,94 @@
 
 package org.apache.spark.sql.execution.streaming
 
+import java.util.concurrent.TimeUnit
+
+import scala.concurrent.duration.Duration
+
 import org.apache.spark.annotation.{Evolving, Experimental}
 import org.apache.spark.sql.streaming.Trigger
+import org.apache.spark.unsafe.types.CalendarInterval
+
+private[sql] object TriggerIntervalUtils {
```

Comment:
Just `object Triggers`? that's the name of the file after all
Do these even need to be moved into their own object? they seem like they're only used in particular classes in this file and aren't abstracting a lot.

Reply:
Yeah, renaming is not a big deal. I'll do it once we have decision on below.

Regarding necessity of this object, it abstracts `interval` to provide unified way to convert any kind of intervals what we support, to the internal format of interval. If we want to support other kinds of interval we can add it here, and any new trigger implementations which leverages interval should have benefit on this.

I also hope if we can avoid having same kind of methods among Trigger, companion object of each implementation, and this object. I thought it is just simpler to have this as utility object, but if we want better shape I'll see whether it can be, maybe `trait` would work?

Reply:
Yeah OK they're reused across the implementation in this file. Seems OK but I think you can declare it `private` even? I wasn't getting at a `trait` or anything.

Reply:
> Seems OK but I think you can declare it private even?

Yeah I think that could be. I'll make a change.